storage/paths/index.go

```diff
@@ -61,6 +64,87 @@ type storageEntry struct {
 	heartbeatErr  error
 }
 
+type IndexProxy struct {
```

Comment:
Would be nice to move the proxy / DB impl to separate files before this is merged

Reply:
For testing could also ask chatgpt to write a version of the proxy which would call both backends and compare results.

Reply:
> Would be nice to move the proxy / DB impl to separate files before this is merged
Yep, thats part of my TODOs

> For testing could also ask chatgpt to write a version of the proxy which would call both backends and compare results
The problem here is how do we get a bool to decide whats the primary path without actually modifying the API